### ISSUE TITLE
SandboxMirror.kext fails to compile on macOS 10.12.4 with XCode 8.1 and up

### ISSUE THREAD

**Issue by feicong** [steven-michaud/SandboxMirror#3]
my xcode version is 8.3.2 (8E2002)，use `xcodebuild` to compile sandboxmirrord ok，but compile SandboxMirror generated the following error:
```
SandboxMirror/SandboxMirror/SandboxMirror/SandboxMirror.cpp:6574:5: error:
      no member named 'mpo_reserved31' in
      'mac_policy_ops'
    SET_HOOK_OVERRIDE_RES(system_check_sysctl, reserved31);
```

**Comment by steven-michaud** [comment]
I just had no problem building on macOS 10.12.4 with XCode 8.0 (8A218a).  Let me try again with later versions and see what happens.  It's possible they're at fault here.

**Comment by steven-michaud** [comment]
I've confirmed your report.  I installed XCode 8.3.2 (8E2002) over my previous copy of XCode 8.0, and I now get exactly the same error as you using xcodebuild (from the command line).

It will take me a while to figure out how to work around this.  In the meantime you could try installing XCode 8.0 on your system and using that.  You'll need to install it under a different name (or in a different directory).  But once you have two versions of XCode installed, you can use xcode-select to choose which is active.

**Comment by feicong** [comment]
thx, i'll try.

**Comment by steven-michaud** [comment]
By trial and error, I've found out that XCode 8.0 is the last (most recent) version of XCode that works properly to compile SandboxMirror.kext.  I'll look for a way to work around this, but (as I said) that may take a while.

In the meantime you should use XCode 8.0 to compile SandboxMirror.kext on macOS 10.12.

**Comment by steven-michaud** [comment]
This was suprisingly easy to fix.  It turns out that as of XCode 8.1, Apple renumbered the "mpo_reservedNN" members of "struct mac_policy_ops" in the macOS 10.12 SDK's mac_policy.h.  Fortunately Apple increments MAC_POLICY_OPS_VERSION whenever it changes this structure.  So we can check its value to see what kind of mpo_reservedNN numbering to expect.

Please download the current master branch and try compiling it with XCode 8.3.2.

**Comment by steven-michaud** [comment]
As best I can tell, this is fixed.
